snakemake/script/__init__.py

```
@@ -112,6 +112,90 @@ def log_fmt_shell(
+                        params[i] = pl.from_dict(value).lazy()
+                    elif param_type == "pl.DataFrame":
+                        params[i] = pl.from_dict(value)
+                    elif param_type == "pl.Series":
+                        params[i] = pl.Series(**value)
+        except ImportError as e:
+            raise ImportError(
+                "Failed to import required module for loading rule params. "
+                "Make sure that the respective package (numpy, pandas, polars) "
+                "is available in the software environment in which the "
+                f"script/wrapper/notebook is executed: {e}"
+            )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)